app/models/post.rb

```diff
@@ -1,3 +1,16 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :update_user_posts_counter
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_user_posts_counter
+    author.update_posts_counter
```

Comment:
- [ ] This method should update the posts counter for a user but currently, it's not doing anthing. Please make sure to fix this issue.
  - You can consider using the below code:
  
```ruby
private
  def update_posts_counter
     author.update(posts_counter: posts.count)
  end